tests/dags/providers/test_factory_utils.py

```diff
@@ -0,0 +1,210 @@
+    instance initialization, *not* while calling ingest_records.
+
+    This also effectively checks that ingest_records does not receive the `*args` passed
+    into pull_media_wrapper, since this lambda doesn't accept any arguments!
+    """
+    fdi.ingest_records = lambda: mock_func(value)
+    return fdi
+
+
+# We have to pass a class down into the various functions, but we want to access
+# entities inside the produced object (e.g. stores) in order to test that they
+# were altered correctly. Best way to do that is to set up a mock that just returns
+# the class when called.
+FakeDataIngesterClass = mock.MagicMock()
+FakeDataIngesterClass.__name__ = "FakeDataIngesterClass"
+FakeDataIngesterClass.side_effect = _set_up_ingester
+
+
+@pytest.mark.parametrize(
+    "func, media_types, stores",
+    [
+        # Happy path
+        (fake_provider_module.main, ["image"], [fake_provider_module.image_store]),
+        # Empty case, no media types provided
+        (fake_provider_module.main, [], []),
+        # Provided function doesn't have a store at the module level
+        pytest.param(
+            requests.get,
+            ["image"],
+            [None],
+            marks=pytest.mark.raises(
+                exception=ValueError, match="Expected stores in .*? were missing.*"
+            ),
+        ),
+        # Provided function doesn't have all specified stores
+        pytest.param(
+            fake_provider_module.main,
+            ["image", "other"],
+            [None, None],
+            marks=pytest.mark.raises(
+                exception=ValueError, match="Expected stores in .*? were missing.*"
+            ),
+        ),
+    ],
+)
+def test_load_provider_script(func, media_types, stores):
+    actual_stores = factory_utils._load_provider_script_stores(
+        func,
+        media_types,
+    )
+    assert actual_stores == dict(zip(media_types, stores))
+
+
+@pytest.mark.parametrize(
+    "func, media_types, stores",
+    [
+        ##################################################
+        # Function-based
+        ##################################################
+        # Happy path
+        (fake_provider_module.main, ["image"], [fake_provider_module.image_store]),
+        # Multiple types
+        (
+            fake_provider_module.main,
+            ["image", "audio"],
+            [fake_provider_module.image_store, fake_provider_module.audio_store],
+        ),
+        # Empty case, no media types provided
+        (fake_provider_module.main, [], []),
+        ##################################################
+        # Class-based
+        ##################################################
+        # Happy path
+        (FakeDataIngesterClass, ["image", "audio"], list(fdi.media_stores.values())),
+        # No media types provided, ingester class still supplies stores
+        (FakeDataIngesterClass, 2, list(fdi.media_stores.values())),
```

Comment:
This isn't a reachable code path, right? Just proving that when an IngesterClass is supplied it doesn't matter what value is passed in for the `media_stores`, since it isn't used?

Reply:
That's right! I assume these tests will be altered when we can simplify all this down and these will probably go away 🙂